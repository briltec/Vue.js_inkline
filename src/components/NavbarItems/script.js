import ITransitionExpand from '@inkline/inkline/transitions/TransitionExpand';
import { breakpoints } from "@inkline/inkline/constants";

export default {
    name: 'INavbarItems',
    inject: [ 'navbar' ],
    components: {
        ITransitionExpand
    },
    data() {
        return {
            collapsible: false
        }
    },
    methods: {
        onWindowResize() {
            this.collapsible = window.innerWidth <= breakpoints[this.navbar.collapse][1];
        }
    },
    created () {
        window.addEventListener('resize', this.onWindowResize);
        this.onWindowResize();
    },
    destroyed() {
        window.removeEventListener('resize', this.onWindowResize)
    }
};