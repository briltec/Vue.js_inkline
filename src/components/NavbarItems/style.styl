@import '../../css/config'
@import '../../css/mixins'
@import '../Navbar/css/_variables'
@import 'css/_variants'

/**
 * Navbar Items
 */
.navbar > .container > .items
    display: flex
    flex-wrap: wrap

    @extend .navbar.-light > .container > .items

