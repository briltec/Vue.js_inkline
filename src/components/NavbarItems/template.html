<div class="items"
     :class="classes">
    <slot></slot>
</div>