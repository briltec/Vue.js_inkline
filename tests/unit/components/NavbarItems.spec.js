import { shallowMount } from '@vue/test-utils';
import NavbarItems from '@inkline/inkline/src/components/NavbarItems';

describe('Components', () => {
    describe('NavbarItems', () => {
        let wrapper;

        beforeEach(() => {
            wrapper = shallowMount(NavbarItems, {
                provide: {
                    navbar: {
                        collapsed: false,
                        collapse: 'md'
                    }
                },
                methods: {
                    created: NavbarItems.created,
                    destroyed: NavbarItems.destroyed
                }
            });
        });

        it('should render correctly', () => {
            expect(wrapper.html()).toMatchSnapshot();
        });

        describe('data()', () => {
            describe('collapsible', () => {
                it('should be defined', () => {
                    expect(wrapper.vm.collapsible).toEqual(false);
                });
            });
        });

        describe('methods', () => {
            describe('onWindowResize()', () => {
                it('should be defined', () => {
                    expect(wrapper.vm.onWindowResize).toBeDefined();
                });

                it('should set collapsible to true if window.innerWidth is less than or equal to breakpoint maximum', () => {
                    window.innerWidth = 600;
                    wrapper.vm.onWindowResize();
                    expect(wrapper.vm.collapsible).toEqual(true);
                });

                it('should set collapsible to false if window.innerWidth is greater than breakpoint maximum', () => {
                    window.innerWidth = 1000;
                    wrapper.vm.onWindowResize();
                    expect(wrapper.vm.collapsible).toEqual(false);
                });
            });
        });

        describe('created()', () => {
            it('should add window resize event listener', () => {
                const spy = jest.spyOn(window, 'addEventListener');

                wrapper.vm.created();

                expect(spy).toHaveBeenCalled();
                expect(spy).toHaveBeenCalledWith('resize', wrapper.vm.onWindowResize);
            });

            it('should call onWindowResize', () => {
                const spy = jest.spyOn(wrapper.vm, 'onWindowResize');

                wrapper.vm.created();

                expect(spy).toHaveBeenCalled();
            });
        });

        describe('destroyed()', () => {
            it('should remove window resize event listener', () => {
                const spy = jest.spyOn(window, 'removeEventListener');

                wrapper.vm.destroyed();

                expect(spy).toHaveBeenCalled();
                expect(spy).toHaveBeenCalledWith('resize', wrapper.vm.onWindowResize);
            });
        });
    });
});
