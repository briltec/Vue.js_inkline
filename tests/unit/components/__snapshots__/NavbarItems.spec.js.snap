// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components NavbarItems should render correctly 1`] = `<div class="items"></div>`;
